internal/tmpl/tmpl.go

```diff
@@ -64,6 +65,7 @@ func (t *Template) WithArtifact(a artifact.Artifact, replacements map[string]str
 	t.fields[arch] = replace(replacements, a.Goarch)
 	t.fields[arm] = replace(replacements, a.Goarm)
 	t.fields[binary] = bin
+	t.fields[archive] = a.Name
```

Comment:
I think the key should probably be `Artifact` or `ArtifactName`, as it can be other kind of artifacts, not only an archive...